include/thingset.h

```diff
@@ -1699,6 +1709,27 @@ int thingset_process_message(struct thingset_context *ts, const uint8_t *msg, si
 int thingset_export_subsets(struct thingset_context *ts, uint8_t *buf, size_t buf_size,
                             uint16_t subsets, enum thingset_data_format format);
 
+#ifdef CONFIG_THINGSET_PROGRESSIVE_IMPORT_EXPORT
+/**
+ * Exports object data for the given subset to the supplied buffer in the specified format starting
+ * at the given object index. At present, only binary formats are supported.
+ *
+ * @param ts Pointer to ThingSet context.
+ * @param buf Pointer to the buffer where the data should be stored
+ * @param buf_size Size of the buffer, i.e. maximum allowed length of the data
+ * @param subsets Flags to select which subset(s) of data items should be exported
+ * @param format Protocol data format to be used (text, binary with IDs or binary with names)
+ * @param index Pointer to an integer which tracks the current object being exported
+ * @param size Indicates the size of the data currently in the buffer
```

Comment:
This comment does not make the purpose of the size parameter completely clear to me and the name size could be confused with `buf_size`. Suggestion:
```suggestion
 * @param len Number of bytes written to the buffer.
```

Alternatively, the number of bytes could just be the return value of the function instead of `1` for successful export.